modules/services/ihaskell.nix

```diff
@@ -0,0 +1,78 @@
+# Adapted from Nixpkgs.
+
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.ihaskell;
+  ihaskellSrc = pkgs.fetchFromGitHub {
+    owner = "gibiansky";
+    repo = "IHaskell";
+    rev = "c2cb8e6789c3f2485ba9e1e0436177b700edb227";
+    sha256 = "0h5hpdfa3daabf1ijl7y5qnpy60yrf7h0ws2n3v5przfmgr8phv5";
+  };
+  ihaskellResult = (import (ihaskellSrc + /release.nix) {
+    compiler = "ghc864";
```

Comment:
I don't think hardcoding the compiler version here like this is a great idea. I would prefer to have it be configurable, but default to `ghc84` (or the latest available version in nixpkgs).